generators/server/templates/src/main/java/package/web/rest/_AccountResource.java

```diff
@@ -274,140 +282,149 @@ public User getAccount(Principal principal) {
-   public void saveAccount(@Valid @RequestBody UserDTO userDTO) {
-       final String userLogin = SecurityUtils.getCurrentUserLogin();
-       Optional<User> existingUser = userRepository.findOneByEmailIgnoreCase(userDTO.getEmail());
-       if (existingUser.isPresent() && (!existingUser.get().getLogin().equalsIgnoreCase(userLogin))) {
-           throw new EmailAlreadyUsedException();
-       }
-       userRepository.findOneByLogin(userLogin).orElseThrow(RuntimeException::new);
-       userService.updateUser(userDTO.getFirstName(), userDTO.getLastName(), userDTO.getEmail(),
-                   userDTO.getLangKey()<% if (databaseType === 'mongodb' || databaseType === 'sql') { %>, userDTO.getImageUrl()<% } %>);
+    @PostMapping("/account")
+    @Timed
+    public void saveAccount(@Valid @RequestBody UserDTO userDTO) {
+        final String userLogin = SecurityUtils.getCurrentUserLogin();
+        Optional<User> existingUser = userRepository.findOneByEmailIgnoreCase(userDTO.getEmail());
+        if (existingUser.isPresent() && (!existingUser.get().getLogin().equalsIgnoreCase(userLogin))) {
+            throw new EmailAlreadyUsedException();
+        }
+        Optional<User> user = userRepository.findOneByLogin(userLogin);
+        if (!user.isPresent()) {
+            throw new DefaultException("The user could not be found.");
+        }
+        userService.updateUser(userDTO.getFirstName(), userDTO.getLastName(), userDTO.getEmail(),
+            userDTO.getLangKey()<% if (databaseType === 'mongodb' || databaseType === 'sql') { %>, userDTO.getImageUrl()<% } %>);
    }
 
-   /**
+    /**
     * POST  /account/change-password : changes the current user's password
     *
     * @param password the new password
     * @throws InvalidPasswordException 400 (Bad Request) if the new password is incorrect
     */
-   @PostMapping(path = "/account/change-password")
-   @Timed
-   public void changePassword(@RequestBody String password) {
-       if (!checkPasswordLength(password)) {
-           throw new InvalidPasswordException();
-       }
-       userService.changePassword(password);
+    @PostMapping(path = "/account/change-password")
+    @Timed
+    public void changePassword(@RequestBody String password) {
+        if (!checkPasswordLength(password)) {
+            throw new InvalidPasswordException();
+        }
+        userService.changePassword(password);
    }<% if (authenticationType === 'session') { %>
 
-   /**
+    /**
     * GET  /account/sessions : get the current open sessions.
     *
     * @return the current open sessions
     * @throws RuntimeException 500 (Internal Server Error) if the current open sessions couldn't be retrieved
     */
-   @GetMapping("/account/sessions")
-   @Timed
-   public List<PersistentToken> getCurrentSessions() {
-       return persistentTokenRepository.findByUser(
-           userRepository.findOneByLogin(SecurityUtils.getCurrentUserLogin())
-               .orElseThrow(RuntimeException::new)
-       );
```

Comment:
`.orElseThrow(() -> new DefaultException("User could not be found"));` would be better